components/script/dom/webglrenderingcontext.rs

```diff
@@ -1035,17 +1039,38 @@ impl WebGLRenderingContext {
     }
 
     fn get_gl_extensions(&self) -> String {
-        let (sender, receiver) = webrender_api::channel::msg_channel().unwrap();
-        self.ipc_renderer
-            .send(CanvasMsg::WebGL(WebGLCommand::GetExtensions(sender)))
-            .unwrap();
+        let (sender, receiver) = webgl_channel().unwrap();
+        self.send_command(WebGLCommand::GetExtensions(sender));
         receiver.recv().unwrap()
     }
+
+    fn handle_layout(&self) -> webrender_api::ImageKey {
```

Comment:
nit: `handle_layout` seems like it's handling some kind of command or what not. Instead it's _returning_ a handle. What about renaming it to `layout_handle`?